win32ss/user/ntuser/menu.c

```diff
@@ -3023,7 +3023,14 @@ static BOOL FASTCALL MENU_ShowPopup(PWND pwndOwner, PMENU menu, UINT id, UINT fl
         if ((y - height) < monitor->rcMonitor.top || y >= monitor->rcMonitor.bottom)
             y = monitor->rcMonitor.bottom - height;
         else
-            y -= height;
+        {
+            adjHgt = y + UserGetSystemMetrics(SM_CYMENUSIZE) +
+                     2 * UserGetSystemMetrics(SM_CYDLGFRAME);
+            if (adjHgt >= monitor->rcMonitor.bottom)
+                y -= height;
+            else
+                y = adjHgt - height;
+        }
```

Comment:
I wonder what you would get (with a screenshot) by adding (or enabling an existing)
`#define SHOW_DEBUGRECT`
in this file, and adding just after line 2990 (i.e. just after ` monitor = UserMonitorFromPoint( ptx, MONITOR_DEFAULTTONEAREST );`), the following:
```c
#if SHOW_DEBUGRECT
    {
        RECT rr = {x, y, x + width, y + height};
        DebugRect(&rr, RGB(0, 255, 255));
    }
#endif
```

Reply:
Here are some examples for you by adding your proposed code above and enabling it:

![@HBelusca_Test_01](https://github.com/reactos/reactos/assets/32620577/f6c4b566-d7ae-4230-8bc6-0f34f61b490e)

![@HBelusca_Test_02](https://github.com/reactos/reactos/assets/32620577/1dcf9898-6f8f-4351-a1c3-4c98b684ec5a)

![@HBelusca_Test_03](https://github.com/reactos/reactos/assets/32620577/2cff5509-00a8-4c8a-bf81-6ebbbedb44d8)

Reply:
Thanks